osu.Game.Rulesets.Osu.Tests/TestSceneSpinner.cs

```diff
@@ -17,32 +17,54 @@ public class TestSceneSpinner : OsuSkinnableTestScene
     {
         private int depthIndex;
 
-        public TestSceneSpinner()
+        private TestDrawableSpinner drawableSpinner;
+
+        [TestCase(false)]
+        [TestCase(true)]
+        public void TestVariousSpinners(bool autoplay)
         {
             AddStep("Miss Big", () => SetContents(() => testSingle(2)));
             AddStep("Miss Medium", () => SetContents(() => testSingle(5)));
             AddStep("Miss Small", () => SetContents(() => testSingle(7)));
-            AddStep("Hit Big", () => SetContents(() => testSingle(2, true)));
-            AddStep("Hit Medium", () => SetContents(() => testSingle(5, true)));
-            AddStep("Hit Small", () => SetContents(() => testSingle(7, true)));
+            AddStep("Hit Big", () => SetContents(() => testSingle(2, autoplay)));
+            AddStep("Hit Medium", () => SetContents(() => testSingle(5, autoplay)));
+            AddStep("Hit Small", () => SetContents(() => testSingle(7, autoplay)));
+        }
+
+        [TestCase(false)]
+        [TestCase(true)]
+        public void TestLongSpinner(bool autoplay)
+        {
+            AddStep("Very short spinner", () => SetContents(() => testSingle(5, autoplay, 2000)));
+            AddUntilStep("Wait for completion", () => drawableSpinner.Result.HasResult);
+            AddUntilStep("Check correct progress", () => drawableSpinner.Progress == (autoplay ? 1 : 0));
+        }
+
+        [TestCase(false)]
+        [TestCase(true)]
+        public void TestSuperShortSpinner(bool autoplay)
+        {
+            AddStep("Very short spinner", () => SetContents(() => testSingle(5, autoplay, 200)));
+            AddUntilStep("Wait for completion", () => drawableSpinner.Result.HasResult);
+            AddUntilStep("Short spinner implicitly completes", () => drawableSpinner.Progress == 1);
         }
 
-        private Drawable testSingle(float circleSize, bool auto = false)
+        private Drawable testSingle(float circleSize, bool auto = false, double length = 3000)
         {
-            var spinner = new Spinner { StartTime = Time.Current + 2000, EndTime = Time.Current + 5000 };
+            var spinner = new Spinner { StartTime = Time.Current + 2000, EndTime = Time.Current + +2000 + length };
```

Comment:
Not sure what the unary `+` is for:

```suggestion
            var spinner = new Spinner { StartTime = Time.Current + 2000, EndTime = Time.Current + 2000 + length };
```

Reply:
typo